src/components/Explore.tsx

```diff
@@ -0,0 +1,263 @@
+  });
+
+  if (error) {
+    console.error("An unexpected error occurred: " + error);
+  }
+
+  if (isLoading) {
+    return (
+      <div className="search-comic-container w-full h-3/10 flex flex-col items-center">
+        <GiSpiderWeb size={70} className="text-white" />
+        <h2 className="font-semibold text-white text-xl text-center">
+          Building the web
+        </h2>
+      </div>
+    );
+  }
+
+  if (
+    !(
+      searchComicByNameResults?.[0]?.md_covers?.[0]?.b2key ||
+      searchComicByNameResults?.[0]?.slug ||
+      searchComicByNameResults?.[0]?.title
+    )
+  ) {
+    return (
+      <div className="w-full h-3/10 text-center">
+        <p className="p-2 font-sans text-white">
+          There is no comics related to {comicName} to display.
+        </p>
+      </div>
+    );
+  }
+
+  return <SearchResults results={searchComicByNameResults} />;
+}
+function SearchComicsByGenreComponent({ genre }: { genre: string }) {
+  const {
+    data: searchComicByGenreResults,
+    error,
+    isLoading,
+  } = useQuery({
+    queryKey: [`searchComicByGenre`, genre],
+    queryFn: () => searchComicByGenre(genre),
+  });
+
+  if (error) {
+    console.error("An unexpected error occurred: " + error);
+  }
+
+  if (isLoading) {
+    return (
+      <div
+        className={`${genre}-container w-full h-3/10 flex flex-col items-center`}
+      >
+        <GiSpiderWeb size={70} className="text-white" />
+        <h2 className="font-semibold text-white text-xl text-center">
+          Building the web
+        </h2>
+      </div>
+    );
+  }
+
+  if (
+    !(
+      searchComicByGenreResults?.[0]?.md_covers?.[0]?.b2key ||
+      searchComicByGenreResults?.[0]?.slug ||
+      searchComicByGenreResults?.[0]?.title
+    )
+  ) {
+    return (
+      <div className="w-full h-3/10 text-center">
+        <p className="p-2 font-sans text-white">
+          There is no {genre} comics to display.
+        </p>
+      </div>
+    );
+  }
+
+  return <SearchResults results={searchComicByGenreResults} />;
+}
+
+function SearchResults({ results }: { results: SearchComicContent[] }) {
+  const filteredComics: SearchComicContent[] = results
+    .filter((comic: SearchComicContent) => comic.content_rating === "safe")
+    .slice(0, 10);
+
+  return (
+    <>
+      <h2 className="font-semibold text-white text-lg">Search Results</h2>
+      <div className="search-container w-full h-3/10 carousel carousel-center space-x-4 bg-neutral rounded-lg bg-transparent overflow-y-hidden flex align-center py-4 mb-4">
+        {filteredComics?.map((searchResult: SearchComicContent) => (
+          <div
+            key={searchResult.slug}
+            className="carousel-item w-2/6 h-full p-0 m-0 flex flex-col justify-center relative first:pl-0 hover:cursor-pointer md:w-1/10 lg:w-1/10"
+            onClick={() =>
+              (
+                document.getElementById(
+                  `search-comic-popup-${searchResult.slug}`
```

Comment:
Safety checks on `searchResults` and it's nested properties. Same with everything below ⬇️